app/scripts/controllers/app-state.js

```diff
@@ -369,4 +375,31 @@ export default class AppStateController extends EventEmitter {
       serviceWorkerLastActiveTime,
     });
   }
+
+  _createApprovalRequest() {
+    this._approvalRequestId = uuid();
+
+    this._messenger.call(
+      'ApprovalController:addRequest',
+      {
+        id: this._approvalRequestId,
+        origin: this._messenger.controllerName,
```

Comment:
I've since found that the `origin` is either the host of the dApp that started the flow, or `metamask` to imply it was internal. Don't think a constant exists for it but worth checking.